src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -805,12 +853,12 @@ public Pattern getIssuePattern() {
      * Gets the list of project IDs in this Jira.
      * This information could be bit old, or it can be null.
      */
-    public Set<String> getProjectKeys() {
+    protected Set<String> getProjectKeys(Item item) {
```

Comment:
why changing to protected?

Reply:
Well, we have a choice here. I believe that those are helper methods only for this plugin's usage.
Since it was *public* and I'm changing method signature, I could either make this *deprecated* or change it to protected (well, deprecation should be there anyway). I'm open to suggestions though

Reply:
why breaking backward compatibility?

Reply:
`Item item` is added, feel free to propose alternative approach that wouldn't require breaking compatibility.

Reply:
I'm only asking to keep `public Set<String> getProjectKeys()`  and manage internally Item is null.

Reply:
Yes, I know and that's my point. We can't have `item` being null because that would mean we'd introduce (keep) the bug of trying to use wrong credentials, i.e. I can immediately think about a test that would fail on this.
Hence IMHO we cannot keep `public Set<String> getProjectKeys()` without adding another method that is actually having Item parameter.
Or we would need to store the Item property on JiraSite class level, make a new public setter that would expose this new approach to public, but not sure if that's the best approach - exactly why I wanted your opinion in the PR :)

Reply:
keep the previous method (marking it deprecated even log something if it's used?) and add the new one